resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -74,10 +76,11 @@ package object config {
     .doc("Whether to populate Hadoop classpath from `yarn.application.classpath` and " +
       "`mapreduce.application.classpath` Note that if this is set to `false`, it requires " +
       "a `with-Hadoop` Spark distribution that bundles Hadoop runtime or user has to provide " +
-      "a Hadoop installation separately.")
+      "a Hadoop installation separately. By default, for `with-hadoop` Spark distribution, " +
+      "this is set to `false`; for `no-hadoop` distribution, this is set to `true`.")
     .version("2.4.6")
     .booleanConf
-    .createWithDefault(true)
+    .createWithDefault(isHadoopProvided())
```

Comment:
I think it's better to update the migration guide at `core-migration-guide.md`.

Reply:
I also have thought about adding to migration guide. No yarn related migration guide, so maybe can put in core.

Reply:
I'll add it to migration guide as well.

Reply:
I'll add it as a followup PR.

Reply:
@dbtsai, seems like it has not been added yet. Could we add it soon before we forget :-)?

Reply:
Gentle ping.